### PR TITLE
Added Javadocs to remaining classes.

### PR DESCRIPTION
**Pull request by PAndaContron**
Javadocs have been added to:

- BookcaseComponent
- BookComponent
- BookRecipeComponent
- DefaultDocumentData
- PageEditor
- RecipeParagraph

### CODE REVIEW

File: src/main/java/org/terasology/books/DefaultDocumentData.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/DefaultDocumentData.java:38]: You need to add `@param` annotations describing each parameter required by the method/constructor.

File: src/main/java/org/terasology/books/DefaultDocumentData.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/DefaultDocumentData.java:67]: Parameter description required

File: src/main/java/org/terasology/books/DefaultDocumentData.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/DefaultDocumentData.java:76]: Parameter description required

File: src/main/java/org/terasology/books/RecipeParagraph.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java]: This description doesn't actually describe what the method does. You should describe the method as something that gets the paragraph render style. Then in a second sentence you could say something like "note that this paragraph's horizontal alignment will always be centered"

File: src/main/java/org/terasology/books/RecipeParagraph.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java]: I would prefer if you didn't inheritDoc here. For most modules, we don't actually compile the Javadoc. The Javadocs are there mostly for people reading the code. So for someone reading this file, the inheritDoc could be a bit annoying (definitely not descriptive).

File: src/main/java/org/terasology/books/RecipeParagraph.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java:156]: Needs method description

File: src/main/java/org/terasology/books/RecipeParagraph.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java:171]: Method description and `@return` missing

File: src/main/java/org/terasology/books/DefaultDocumentData.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/DefaultDocumentData.java]: Needs `@return` description

File: src/main/java/org/terasology/books/RecipeParagraph.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java:84]: Needs `@return`

File: src/main/java/org/terasology/books/RecipeParagraph.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/RecipeParagraph.java:162]: Needs `@return`

File: src/main/java/org/terasology/books/logic/BookRecipeComponent.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/logic/BookRecipeComponent.java]: Please get rid of the `&lt` and `&gt`. They will probably work if the Javadocs are compiled, but they make for bad readability

File: src/main/java/org/terasology/books/logic/BookRecipeComponent.java
Comment (mjuvekar7) [inline src/main/java/org/terasology/books/logic/BookRecipeComponent.java]: Again, I would prefer if you got rid of the &lt/gt

**Review by mjuvekar7 — CHANGES_REQUESTED** [review]

**Comment by PAndaContron** [comment]
Ok, I will make sure to change these. Most of the tags are missing because I was under the impression that the Javadocs will  be compiled (I use Eclipse so it does that automatically on the side), so they would be inherited. I'll fix it when I get home.

**Comment by Waterpicker** [comment]
Approved after I dealt with a merge conflict.

**Pull request merged**
